runners/s3-benchrunner-rust/src/main.rs

```diff
@@ -0,0 +1,69 @@
+use clap::{Parser, ValueEnum};
+use std::time::Instant;
+
+use s3_benchrunner_rust::{BenchmarkConfig, BenchmarkRunner, TransferManagerRunner};
+
+#[derive(Parser)]
+#[command()]
+struct Args {
+    #[arg(value_enum, help = "ID of S3 library to use")]
+    s3_client: S3ClientId,
+    #[arg(help = "Path to workload file (e.g. download-1GiB.run.json)")]
+    workload: String,
+    #[arg(help = "S3 bucket name (e.g. my-test-bucket)")]
+    bucket: String,
+    #[arg(help = "AWS Region (e.g. us-west-2)")]
+    region: String,
+    #[arg(help = "Target throughput, in gigabits per second (e.g. \"100.0\" for c5n.18xlarge)")]
+    target_throughput: f64,
+}
+
+#[derive(ValueEnum, Clone)]
+enum S3ClientId {
+    #[clap(name = "sdk-rust-tm", help = "use aws-s3-transfer-manager crate")]
+    TransferManager,
+    // TODO:
+    // #[clap(name="sdk-rust-client", help="use aws-sdk-s3 crate")]
+    // SdkClient,
+}
+
+fn main() {
+    let args = Args::parse();
+
+    let config = BenchmarkConfig::new(
+        &args.workload,
+        &args.bucket,
+        &args.region,
+        args.target_throughput,
+    );
+
+    // create appropriate benchmark runner
+    let runner: Box<dyn BenchmarkRunner> = match args.s3_client {
+        S3ClientId::TransferManager => Box::new(TransferManagerRunner::new(config)),
+    };
+    let workload = &runner.config().workload;
+    let bytes_per_run: u64 = workload.tasks.iter().map(|x| x.size).sum();
```

Comment:
You may be able to simplify this to:
```suggestion
    let bytes_per_run = workload.tasks.iter().map(u64::size).sum();
```

Reply:
This doesn't compile?
I'm summing up all the `Task.size` in this array of `Task` structs